src/Events/Category_Colors/Migration/Utilities.php

```diff
@@ -0,0 +1,317 @@
+		],
+		'reset_show'               => [
+			'mapped_key' => 'category-color-reset-button',
+			'validation' => 'boolean',
+			'import'     => true,
+		],
+		'show_ignored_cats_legend' => [
+			'mapped_key' => 'category-color-show-hidden-categories',
+			'validation' => 'boolean',
+			'import'     => true,
+		],
+	];
+
+
+	/**
+	 * Prefix for storing category meta values.
+	 *
+	 * @since TBD
+	 * @var string
+	 */
+	protected string $meta_key_prefix = 'tec-events-cat-colors-';
+
+	/**
+	 * Mapping of old meta keys to new ones.
+	 * Keys represent the old names, and values represent the new names.
+	 * Any key not in this list will be ignored.
+	 *
+	 * @since TBD
+	 * @var array<string, string>
+	 */
+	protected array $meta_key_map = [
+		'border'     => 'primary',
+		'background' => 'secondary',
+		'text'       => 'text',
+	];
+
+	/**
+	 * Retrieves the original settings from the database.
+	 *
+	 * @since TBD
+	 * @return array<string, mixed> The original settings.
+	 */
+	public function get_original_settings(): array {
+		return get_option( $this->original_settings_option, [] );
+	}
+
+	/**
+	 * Retrieves the migration data from the database.
+	 *
+	 * @since TBD
+	 * @return array<string, mixed> The migration data.
+	 */
+	public function get_migration_data(): array {
+		return get_option( $this->migration_data_option, [] );
+	}
+
+	/**
+	 * Stores the migration data in the database.
+	 *
+	 * @since TBD
+	 *
+	 * @param array<string, mixed> $data The processed migration data to store.
+	 *
+	 * @return void
+	 */
+	public function update_migration_data( array $data ): void {
+		update_option( $this->migration_data_option, $data, false );
+	}
+
+	/**
+	 * Extracts the category ID from a category-related setting key.
+	 *
+	 * @since TBD
+	 *
+	 * @param string $key The category setting key.
+	 *
+	 * @return int|null The extracted category ID or null if not found.
+	 */
+	protected function extract_category_id( string $key ): ?int {
+		$original_settings = $this->get_original_settings();
+
+		// Check if the key exists in the terms or all_terms mapping.
+		foreach ( [ 'terms', 'all_terms' ] as $term_group ) {
+			if ( isset( $original_settings[ $term_group ] ) ) {
+				foreach ( $original_settings[ $term_group ] as $term_id => [$slug, $name] ) {
+					if ( strpos( $key, $slug ) === 0 ) {
+						return (int) $term_id;
+					}
+				}
+			}
+		}
+
+		return null;
+	}
+
+	/**
+	 * Retrieves the mapped meta key, or null if it is not recognized.
+	 *
+	 * @since TBD
+	 *
+	 * @param string $key The original meta key.
+	 *
+	 * @return string|null The mapped meta key, or null if it should be ignored.
+	 */
+	protected function get_mapped_meta_key( string $key ): ?string {
+		return $this->meta_key_map[ $key ] ?? null;
+	}
+
+	/**
+	 * Gets the current migration status.
+	 *
+	 * @since TBD
+	 * @return array<string, mixed> The current migration status with timestamp.
+	 */
+	protected function get_migration_status(): array {
+		return get_option(
```

Comment:
You should cast this to ensure there's no possible way for the function to return something other than an array.

```suggestion
		return (array) get_option(
```